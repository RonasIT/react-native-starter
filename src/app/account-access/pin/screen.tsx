import { RouteProp } from '@react-navigation/native';
import React, { ReactElement, useEffect, useState } from 'react';
import { LocalAuthActions } from '@shared/local-auth';
import { localAuthService } from '@shared/local-auth/service';
import { PinForm } from '@shared/pin-form/component';
import { storeRef } from '@store';
import { AccountAccessNavigationParams } from '../navigation';

const pinLength = 4;

export function PinScreen(props: { route: RouteProp<AccountAccessNavigationParams, 'Pin'> }): ReactElement {
  const isPinSet = props?.route?.params?.isPinSet;
  const [inputValue, setInputValue] = useState('');
  const [confirmPin, setConfirmPin] = useState('');
  const [hasError, setHasError] = useState(false);
  const confirmMode = !isPinSet && inputValue.length === pinLength;

  const checkPinCode = async (): Promise<void> => {
    const isCorrect = await localAuthService.checkPin(inputValue);
    isCorrect ? storeRef.dispatch(LocalAuthActions.localAuthSuccess()) : setHasError(true);
  };

  const handleError = (): void => {
    setInputValue('');
    setConfirmPin('');
    setHasError(false);
  };

  useEffect(() => {
    if (inputValue.length === pinLength && isPinSet) {
      checkPinCode();
    }
    if (confirmMode && confirmPin.length === pinLength) {
      confirmPin === inputValue ? storeRef.dispatch(LocalAuthActions.setPin({ pin: confirmPin })) : setHasError(true);
    }
  }, [inputValue, confirmPin]);

  return (
    <PinForm
      hasError={hasError}
      handleError={handleError}
      pinLength={pinLength}
      value={confirmMode ? confirmPin : inputValue}
      onValueChange={confirmMode ? setConfirmPin : setInputValue}
      title={isPinSet ? 'Enter Pin' : confirmMode ? 'Confirm Pin' : 'Set Pin'}
      isPinSet={isPinSet}
    />
  );
}
