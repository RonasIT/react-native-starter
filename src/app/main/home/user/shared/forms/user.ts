import * as Yup from 'yup';
import { useTranslation } from '@shared/i18n';

export class UserForm {
  public email: string;
  public name: string;
  public gender: string;
  public status: string;

  public static get validationSchema(): Yup.SchemaOf<UserForm> {
    const translate = useTranslation('COMMON.VALIDATION');

    return Yup.object().shape({
      email: Yup.string()
        .email(translate('TEXT_VALIDATION_EMAIL'))
        .required(translate('TEXT_VALIDATION_REQUIRED_FIELD')),
      name: Yup.string().required(translate('TEXT_VALIDATION_REQUIRED_FIELD')),
      gender: Yup.string().required(translate('TEXT_VALIDATION_REQUIRED_FIELD')),
      status: Yup.string().required(translate('TEXT_VALIDATION_REQUIRED_FIELD'))
    });
  }

  constructor() {
    this.email = '';
    this.name = '';
    this.gender = '';
    this.status = '';
  }
}
