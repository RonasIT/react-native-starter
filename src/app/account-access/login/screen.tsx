import { yupResolver } from '@hookform/resolvers/yup';
import Constants from 'expo-constants';
import { isEmpty } from 'lodash';
import React from 'react';
import { useForm } from 'react-hook-form';
import { ScrollView, View } from 'react-native';
import { Image, Keyboard } from 'react-native-ui-lib';
import { assets } from '@assets/index';
import { AppVersion } from '@shared/app-version';
import { AppButton } from '@shared/button';
import { useTranslation } from '@shared/i18n';
import { InputFormGroup } from '@shared/input-form-group';
import { AppScreen } from '@shared/screen';
import { AppText } from '@shared/text';
import { commonStyle, createStyles } from '@styles';
import { loginScreenFacade } from './facade';
import { LoginForm } from './shared/forms';

export function LoginScreen(): JSX.Element {
  const translate = useTranslation('ACCOUNT_ACCESS.LOGIN');
  const { isSubmitting } = loginScreenFacade;
  const appName = Constants.expoConfig.name;

  function formSubmitted(values: LoginForm): void {
    loginScreenFacade.authorize(values);
  }

  const form = useForm<LoginForm>({
    defaultValues: new LoginForm(),
    resolver: yupResolver(LoginForm.validationSchema)
  });
  const { handleSubmit, formState, control } = form;

  return (
    <AppScreen style={commonStyle.container}>
      <ScrollView contentContainerStyle={style.content}>
        <Image source={assets.brand.logo} style={style.logo} />
        <AppText style={style.title} variant='largest'>
          {translate('TEXT_TITLE', { value: appName })}
        </AppText>
        <InputFormGroup<LoginForm>
          label={translate('TEXT_LOGIN')}
          name='email'
          testID='email-input'
          autoCapitalize='none'
          keyboardType='email-address'
          control={control}
        />
        <InputFormGroup<LoginForm>
          isPassword={true}
          testID='password-input'
          label={translate('TEXT_PASSWORD')}
          name='password'
          control={control}
        />
        <View style={style.footer}>
          <AppButton
            isLoading={isSubmitting}
            disabled={!isEmpty(formState.errors) && formState.isSubmitted}
            testID='submit-button'
            label={translate('BUTTON_SUBMIT')}
            onPress={handleSubmit(formSubmitted)}
          />
        </View>
        <AppVersion />
        <Keyboard.KeyboardAwareInsetsView />
      </ScrollView>
    </AppScreen>
  );
}

const style = createStyles({
  content: {
    paddingTop: '2rem'
  },
  logo: {
    width: '3.5rem',
    height: '3.5rem'
  },
  title: {
    marginVertical: '2rem'
  },
  footer: {
    marginTop: '2rem'
  }
});
