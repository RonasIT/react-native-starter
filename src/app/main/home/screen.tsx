import React, { ReactElement } from 'react';
import { AppScreen } from '../../../libs/shared/ui/ui-kit/screen';
import { UsersList } from '../../../libs/users/features/list';

export function HomeScreen(): ReactElement {
  return (
    <AppScreen testID='home-screen'>
      <UsersList />
    </AppScreen>
  );
}
