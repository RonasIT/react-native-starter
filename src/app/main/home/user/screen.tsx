import { RouteProp } from '@react-navigation/native';
import { useFormik } from 'formik';
import React, { useEffect } from 'react';
import { View } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import { HomeNavigationParams } from '@app/main/home/navigation';
import { AppButton } from '@shared/button';
import { useTranslation } from '@shared/i18n';
import { InputFormGroup } from '@shared/input-form-group';
import { AppScreen } from '@shared/screen';
import { AppText, TextTheme } from '@shared/text';
import { userAPI } from '@shared/user/api';
import { commonStyle, createStyles } from '@styles';
import { UserForm } from './shared/forms';

export function UserScreen(props: { route: RouteProp<HomeNavigationParams, 'User'> }): JSX.Element {
  const id = props.route.params?.id;
  const translate = useTranslation('MAIN.USER');
  const { useLazyGetQuery } = userAPI;
  const [loadUser, { data: user }] = useLazyGetQuery();

  useEffect(() => {
    if (id) {
      loadUser({ id });
    }
  }, []);

  useEffect(() => {
    if (user) {
      formik.setValues({
        email: user.email,
        name: user.name
      });
    }
  }, [user]);

  function formSubmitted(values: UserForm): void {
    console.log(values);
  }

  function deleteButtonPressed(): void {
    console.log('Delete');
  }

  const formik = useFormik({
    initialValues: new UserForm(),
    validationSchema: UserForm.validationSchema,
    onSubmit: formSubmitted
  });
  const { handleSubmit } = formik;

  return (
    <KeyboardAwareScrollView>
      <AppScreen style={commonStyle.container}>
        <AppText theme={TextTheme.LARGE} style={style.title}>
          {id ? translate('TEXT_EDIT') : translate('TEXT_CREATE')}
        </AppText>
        <InputFormGroup<UserForm>
          label={translate('TEXT_EMAIL')}
          name='email'
          autoCapitalize='none'
          keyboardType='email-address'
          formik={formik}
        />
        <InputFormGroup
          label={translate('TEXT_NAME')}
          name='name'
          formik={formik} />
        <View style={style.footer}>
          {id && (
            <AppButton
              title={translate('BUTTON_DELETE')}
              onPress={deleteButtonPressed}
              style={[style.button, style.deleteButton]}
              theme='secondary'
            />
          )}
          <AppButton
            isDisabled={!formik.isValid && formik.submitCount > 0}
            title={translate('BUTTON_SAVE')}
            onPress={() => handleSubmit()}
            style={style.button}
          />
        </View>
      </AppScreen>
    </KeyboardAwareScrollView>
  );
}

const style = createStyles({
  title: {
    marginVertical: '2rem'
  },
  footer: {
    marginTop: '1.5rem',
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  button: {
    flex: 1
  },
  deleteButton: {
    marginRight: '1rem'
  }
});
