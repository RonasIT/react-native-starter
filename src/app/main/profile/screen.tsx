import { AppActivityIndicator } from '@shared/activity-indicator';
import { AppButton } from '@shared/button';
import { useTranslation } from '@shared/i18n';
import { AppScreen } from '@shared/screen';
import { AppText, TextTheme } from '@shared/text';
import { commonStyle, createStyles } from '@styles';
import React, { ReactElement } from 'react';
import { profileScreenFacade } from './facade';

export function ProfileScreen(): ReactElement {
  const { profile, logout } = profileScreenFacade;
  const translate = useTranslation('MAIN.PROFILE');

  return (
    <AppScreen style={[commonStyle.container, style.container]}>
      {profile ? (
        <AppText theme={TextTheme.LARGEST}>{translate('TEXT_GREETING', { name: profile?.name })}</AppText>
      ) : (
        <AppActivityIndicator />
      )}
      <AppButton title={translate('BUTTON_LOGOUT')} onPress={logout} />
    </AppScreen>
  );
}

const style = createStyles({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-evenly'
  }
});
