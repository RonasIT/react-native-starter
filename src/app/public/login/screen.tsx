import { AppButton } from '@shared/button';
import { useTranslation } from '@shared/i18n';
import { InputFormGroup } from '@shared/input-form-group';
import { AppScreen } from '@shared/screen';
import { AppText, TextTheme } from '@shared/text';
import { commonStyle, createStyles } from '@styles';
import Constants from 'expo-constants';
import { useFormik } from 'formik';
import React, { useEffect } from 'react';
import { View } from 'react-native';
import { loginScreenFacade } from './facade';
import { LoginForm } from './shared/forms';

export function LoginScreen(): JSX.Element {
  const translate = useTranslation('PUBLIC.LOGIN');
  const { isSubmitting } = loginScreenFacade;
  const appName = Constants.manifest.name;

  function formSubmitted(values: LoginForm): void {
    loginScreenFacade.authorize(values);
  }

  const formik = useFormik({
    initialValues: new LoginForm(),
    validationSchema: LoginForm.validationSchema,
    onSubmit: formSubmitted
  });
  const { handleSubmit } = formik;

  useEffect(() => {
    return () => {
      loginScreenFacade.reset();
    };
  }, []);

  return (
    <AppScreen style={[commonStyle.container, style.container]}>
      <View>
        <AppText style={style.title} theme={TextTheme.LARGEST}>
          {translate('TEXT_TITLE', { value: appName })}
        </AppText>
        <InputFormGroup<LoginForm>
          label={translate('TEXT_LOGIN')}
          name='email'
          testID='email-input'
          autoCapitalize='none'
          keyboardType='email-address'
          formik={formik}
        />
        <InputFormGroup<LoginForm>
          isPassword={true}
          testID='password-input'
          label={translate('TEXT_PASSWORD')}
          name='password'
          formik={formik}
        />
        <View style={style.footer}>
          <AppButton
            isLoading={isSubmitting}
            isDisabled={!formik.isValid && formik.submitCount > 0}
            testID='submit-button'
            title={translate('BUTTON_SUBMIT')}
            onPress={() => handleSubmit()}
          />
        </View>
      </View>
    </AppScreen>
  );
}

const style = createStyles({
  container: {
    flex: 1,
    justifyContent: 'center'
  },
  title: {
    marginBottom: '4rem'
  },
  footer: {
    marginTop: '2rem'
  }
});
