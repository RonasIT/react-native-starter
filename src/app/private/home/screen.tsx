import { homeScreenFacade } from '@app/private/home/facade';
import { ItemsList } from '@shared/items-list';
import { AppScreen } from '@shared/screen';
import { User } from '@shared/user';
import { commonStyle } from '@styles';
import React, { ReactElement, useEffect } from 'react';
import { HomeListItem } from './components';

export function HomeScreen(): ReactElement {
  const { items, isLoading, isRefreshing, pagination } = homeScreenFacade;
  const refreshItems = (): void => homeScreenFacade.refreshItems();
  const loadMore = (): void => homeScreenFacade.loadItems(pagination.currentPage + 1);

  useEffect(() => {
    homeScreenFacade.loadItems();
  }, []);

  return (
    <AppScreen>
      <ItemsList<User>
        data={items}
        renderItem={HomeListItem}
        isLoading={isLoading}
        isRefreshing={isRefreshing}
        canLoadMore={pagination.currentPage < pagination.lastPage}
        onEndReached={loadMore}
        onRefresh={refreshItems}
        containerStyle={commonStyle.container}
        numColumns={1}
      />
    </AppScreen>
  );
}
