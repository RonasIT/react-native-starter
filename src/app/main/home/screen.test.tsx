import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { apiService } from '@shared/api';
import { userService } from '@shared/user';
import { fireEvent, render, RenderAPI, waitFor } from '@testing-library/react-native';
import { userPaginationResponse } from '@tests/fixtures';
import { scrollDownEventData, TestRootComponent } from '@tests/helpers';
import React from 'react';
import { act, ReactTestInstance } from 'react-test-renderer';
import { Observable, of } from 'rxjs';
import { HomeScreen } from './screen';

describe('Home screen', () => {
  let component: RenderAPI;
  let usersList: ReactTestInstance;
  const searchUsersSpy = jest.spyOn(userService, 'search');

  function initComponent(): RenderAPI {
    const { Screen, Navigator } = createStackNavigator();

    return render(
      <TestRootComponent>
        <NavigationContainer>
          <Navigator>
            <Screen name='Home' component={HomeScreen} />
          </Navigator>
        </NavigationContainer>
      </TestRootComponent>
    );
  }

  beforeAll(() => {
    jest.spyOn(apiService, 'get').mockImplementation((endpoint) => {
      if (endpoint === '/users') {
        return of(userPaginationResponse) as Observable<any>;
      }
    });
  });

  beforeEach(() => {
    component = initComponent();
    usersList = component.getByTestId('users-list');
  });

  it('should match the snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render list items', () => {
    const listItems = component.getAllByTestId('user-item');
    expect(listItems).toHaveLength(userPaginationResponse.data.length);
  });

  it('should load more items after the list end reached', async () => {
    fireEvent.scroll(usersList, scrollDownEventData);

    await waitFor(() => {
      expect(searchUsersSpy).toHaveBeenCalledWith({ page: 2 });
    });
  });

  it('should load the first page of users on the list refresh ', async () => {
    const { refreshControl } = usersList.props;
    act(() => {
      refreshControl.props.onRefresh();
    });

    await waitFor(() => {
      expect(searchUsersSpy).toHaveBeenCalledWith({ page: 1 });
    });
  });
});
