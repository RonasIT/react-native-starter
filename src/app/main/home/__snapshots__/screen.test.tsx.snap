// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen should match the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="box-none"
          style={
            Object {
              "zIndex": 1,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            onLayout={[Function]}
            pointerEvents="box-none"
            style={null}
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "rgb(255, 255, 255)",
                    "borderBottomColor": "rgb(216, 216, 216)",
                    "flex": 1,
                    "shadowColor": "rgb(216, 216, 216)",
                    "shadowOffset": Object {
                      "height": 0.5,
                      "width": 0,
                    },
                    "shadowOpacity": 0.85,
                    "shadowRadius": 0,
                  }
                }
              />
            </View>
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "height": 44,
                  "maxHeight": undefined,
                  "minHeight": undefined,
                  "opacity": undefined,
                  "transform": undefined,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "height": 0,
                  }
                }
              />
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "stretch",
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  collapsable={false}
                  pointerEvents="box-none"
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "justifyContent": "center",
                      "marginStart": 0,
                      "opacity": 1,
                    }
                  }
                />
                <View
                  collapsable={false}
                  pointerEvents="box-none"
                  style={
                    Object {
                      "justifyContent": "center",
                      "marginHorizontal": 16,
                      "maxWidth": 718,
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    accessibilityRole="header"
                    aria-level="1"
                    collapsable={false}
                    numberOfLines={1}
                    onLayout={[Function]}
                    style={
                      Object {
                        "color": "rgb(28, 28, 30)",
                        "fontSize": 17,
                        "fontWeight": "600",
                      }
                    }
                  >
                    Home
                  </Text>
                </View>
                <View
                  collapsable={false}
                  pointerEvents="box-none"
                  style={
                    Object {
                      "alignItems": "flex-end",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "justifyContent": "center",
                      "marginEnd": 0,
                      "opacity": 1,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "opacity": 1,
                }
              }
            />
            <View
              accessibilityElementsHidden={false}
              closing={false}
              gestureVelocityImpact={0.3}
              importantForAccessibility="auto"
              onClose={[Function]}
              onGestureBegin={[Function]}
              onGestureCanceled={[Function]}
              onGestureEnd={[Function]}
              onOpen={[Function]}
              onTransition={[Function]}
              pointerEvents="box-none"
              style={
                Array [
                  Object {
                    "display": "flex",
                    "overflow": undefined,
                  },
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                ]
              }
              transitionSpec={
                Object {
                  "close": Object {
                    "animation": "spring",
                    "config": Object {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                  "open": Object {
                    "animation": "spring",
                    "config": Object {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  collapsable={false}
                  forwardedRef={[Function]}
                  handlerTag={1}
                  handlerType="PanGestureHandler"
                  needsOffscreenAlphaCompositing={false}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "backgroundColor": "rgb(242, 242, 242)",
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "flex": 1,
                              },
                              Object {},
                            ]
                          }
                          testID="home-screen"
                        >
                          <RCTScrollView
                            ListEmptyComponent={<ItemsListEmptyState />}
                            ListFooterComponent={false}
                            contentContainerStyle={
                              Array [
                                Object {
                                  "minHeight": "100%",
                                  "paddingBottom": 50,
                                },
                                Object {
                                  "paddingHorizontal": 24,
                                },
                              ]
                            }
                            data={
                              Array [
                                User {
                                  "createdAt": undefined,
                                  "deletedAt": undefined,
                                  "email": "test1@test.com",
                                  "id": 1,
                                  "isDeleting": undefined,
                                  "isUpdating": undefined,
                                  "name": "Test 1",
                                  "updatedAt": undefined,
                                  Symbol(immer-draftable): true,
                                },
                                User {
                                  "createdAt": undefined,
                                  "deletedAt": undefined,
                                  "email": "test2@test.com",
                                  "id": 2,
                                  "isDeleting": undefined,
                                  "isUpdating": undefined,
                                  "name": "Test 2",
                                  "updatedAt": undefined,
                                  Symbol(immer-draftable): true,
                                },
                              ]
                            }
                            getItem={[Function]}
                            getItemCount={[Function]}
                            keyExtractor={[Function]}
                            onContentSizeChange={[Function]}
                            onEndReached={[Function]}
                            onLayout={[Function]}
                            onMomentumScrollBegin={[Function]}
                            onMomentumScrollEnd={[Function]}
                            onScroll={[Function]}
                            onScrollBeginDrag={[Function]}
                            onScrollEndDrag={[Function]}
                            refreshControl={
                              <AppRefreshControl
                                onRefresh={[Function]}
                                refreshing={false}
                              />
                            }
                            removeClippedSubviews={false}
                            renderItem={[Function]}
                            scrollEventThrottle={50}
                            stickyHeaderIndices={Array []}
                            testID="users-list"
                            viewabilityConfigCallbackPairs={Array []}
                          >
                            <RCTRefreshControl />
                            <View>
                              <View
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  style={
                                    Object {
                                      "borderBottomColor": "#FFFFFF",
                                      "borderBottomWidth": 2,
                                      "paddingVertical": 8,
                                    }
                                  }
                                  testID="user-item"
                                >
                                  <Text
                                    numberOfLines={1}
                                    style={
                                      Array [
                                        Object {
                                          "backgroundColor": "transparent",
                                          "color": "#20303C",
                                          "textAlign": "left",
                                        },
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        Array [
                                          Object {
                                            "fontFamily": "SFProDisplayRegular",
                                            "fontSize": 22.4,
                                            "lineHeight": 24.64,
                                          },
                                          Object {
                                            "color": "#FFFFFF",
                                          },
                                          undefined,
                                        ],
                                      ]
                                    }
                                  >
                                    #
                                    1
                                    : 
                                    Test 1
                                  </Text>
                                  <Text
                                    numberOfLines={1}
                                    style={
                                      Array [
                                        Object {
                                          "backgroundColor": "transparent",
                                          "color": "#20303C",
                                          "textAlign": "left",
                                        },
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        Array [
                                          Object {
                                            "fontFamily": "SFProTextRegular",
                                            "fontSize": 16,
                                            "lineHeight": 29.44,
                                          },
                                          Object {
                                            "color": "#FFFFFF",
                                          },
                                          undefined,
                                        ],
                                      ]
                                    }
                                  >
                                    Email: 
                                    test1@test.com
                                  </Text>
                                </View>
                              </View>
                              <View
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  style={
                                    Object {
                                      "borderBottomColor": "#FFFFFF",
                                      "borderBottomWidth": 2,
                                      "paddingVertical": 8,
                                    }
                                  }
                                  testID="user-item"
                                >
                                  <Text
                                    numberOfLines={1}
                                    style={
                                      Array [
                                        Object {
                                          "backgroundColor": "transparent",
                                          "color": "#20303C",
                                          "textAlign": "left",
                                        },
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        Array [
                                          Object {
                                            "fontFamily": "SFProDisplayRegular",
                                            "fontSize": 22.4,
                                            "lineHeight": 24.64,
                                          },
                                          Object {
                                            "color": "#FFFFFF",
                                          },
                                          undefined,
                                        ],
                                      ]
                                    }
                                  >
                                    #
                                    2
                                    : 
                                    Test 2
                                  </Text>
                                  <Text
                                    numberOfLines={1}
                                    style={
                                      Array [
                                        Object {
                                          "backgroundColor": "transparent",
                                          "color": "#20303C",
                                          "textAlign": "left",
                                        },
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        Array [
                                          Object {
                                            "fontFamily": "SFProTextRegular",
                                            "fontSize": 16,
                                            "lineHeight": 29.44,
                                          },
                                          Object {
                                            "color": "#FFFFFF",
                                          },
                                          undefined,
                                        ],
                                      ]
                                    }
                                  >
                                    Email: 
                                    test2@test.com
                                  </Text>
                                </View>
                              </View>
                              <View
                                style={
                                  Object {
                                    "height": 0,
                                  }
                                }
                              />
                            </View>
                          </RCTScrollView>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
